### PR TITLE
Project report updates

### LINKED ISSUE #109: Support for KAPA library prep, Rapid analysis (dragen) and GATK4 pipeline
We need to 
 - [ ] obtain documentation of the KAPA library preparation from the lab,
 - [ ] Document Rapid analysis pipeline
 - [ ] Document GATK4 pipeline

### PR DESCRIPTION
**Pull request by mwhamgenomics**
Adds project report support for samples processed with:

- KAPA library prep
- Rapid analysis with Dragen
- Bioinformatics analysis with GATK4

Closes #109.

### CODE REVIEW

**Comment by lukbut** [comment]
The base branch needs to be updated to the release branch.

File: project_report/project_information.py [hunk 1]
```diff
@@ -230,6 +236,12 @@ def get_project_library_workflows(self):
     def get_project_analysis_types(self):
         return self._aggregate_per_project(self.get_analysis_type_from_sample)
 
+    def has_rapid_samples(self):
+        return any(
+            query_dict(self.sample_info(s['sample_id']), 'info.Rapid Analysis') == 'Yes'
+            for s in self.sample_data_for_project
```
Comment (lukbut) [inline project_report/project_information.py]: Is this correct? It seems unfinished.

File: project_report/report_texts.yaml [hunk 2]
```diff
@@ -52,20 +52,34 @@ library_qc_nano: >-
 library_preparation_pcr_free: >-
   Next Generation sequencing libraries are prepared using Illumina SeqLab specific 
   TruSeq PCR-Free High Throughput library preparation kits in conjunction with the Hamilton MicroLab STAR and 
-  Clarity LIMS X Edition. The gDNA samples are normalised to the concentration and volume required for the Illumina 
+  Clarity LIMS X (4.2) Edition. The gDNA samples are normalised to the concentration and volume required for the Illumina
   TruSeq PCR-Free library preparation kits then sheared to a 450bp mean insert size using a Covaris LE220 
-  focused-ultrasonicator. The inserts are blunt ended, A-tailed, size selected and the TruSeq adapters are ligated onto 
+  focused-ultrasonicator. The inserts are blunt ended, A-tailed, size selected, and the TruSeq adapters are ligated onto
```
Comment (lukbut) [inline project_report/report_texts.yaml]: Ah, the Oxford comma.

File: project_report/project_information.py (hunk 1 above)
Comment (mwhamgenomics) [reply project_report/project_information.py]: Yup, [generator expression](https://docs.python.org/3/reference/expressions.html?highlight=generator#generator-expressions).

File: project_report/project_report_latex.py [hunk 3]
```diff
@@ -362,17 +363,27 @@ def create_method_section(self):
...
-                    with self.doc.create(Subsection('Bioinformatics Analysis', numbering=True)):
-                        add_text(self.doc, report_text.get('bioinformatics_analysis').format(**bioinfo_version))
+                subsection, paragraph = bioinf_reports[bioinfo_analysis_type]
+                with self.doc.create(Subsection(subsection, numbering=True)):
+                    add_text(self.doc, report_text[paragraph].format(**bioinfo_version))
+
+            if self.pi.has_rapid_samples():
+                with self.doc.create(Subsection('Rapid Bioinformatics Analysis', numbering=True)):
+                    add_text(self.doc, report_text['bioinformatics_analysis_rapid'])
+                    self.doc.append(' More information on Dragen can be found on the ')
```
Comment (tcezard) [inline project_report/project_report_latex.py]: Not sure why but the report shows no space between the the end of the previous sentence and `More information`

**Comment by tcezard** [comment]
I've added the support for rapid and gatk in file format and appendix. The document is currently being reviewed in Qpulse.

**Comment by tcezard** [comment]
Document has been accepted in Qpulse, so PR can be merged.

**Pull request merged**